### PR TITLE
issue: 781967 Fix issue with PKG_CHECK_MODULES

### PR DESCRIPTION
**Pull request by igor-ivanov**
Issue description:
The PKG_PROG_PKG_CONFIG variable refers to a macro pkg.m4 that is
provided as part of the pkg-config package, so the first thing to
check is that pkg-config is installed and that the macro file is
in the default location.
Usually it is /usr/share/aclocal
Following command can display folder aclocal uses as default
aclocal --print-ac-dir
If that's not the same as the location above, it suggests there is
a non-standard version of automake on your system - if you can't
resolve that, then a possible workaround is to set or export the
ACLOCAL_PATH environment variable before running the autogen.sh script e.g.
ACLOCAL_PATH=/usr/share/aclocal ./autogen.sh

@OphirMunk please verify and accept

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/100/ for details (Mellanox internal link).

File: configure.ac [hunk 1]
```diff
@@ -19,6 +19,19 @@ AC_CONFIG_SRCDIR(src)
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR(config)
 
+dnl Make sure the pkg-config macros are defined
+m4_ifdef([PKG_PROG_PKG_CONFIG],[],[
```
Comment (miked-mellanox) [inline configure.ac]: btw, maybe it is better to m4_include(m4/pkg.m4) as part of libvma and ensure it is available?

Comment (igor-ivanov) [inline configure.ac]: it is good way. I can do it.

Comment (miked-mellanox) [inline configure.ac]: see example here: https://github.com/openucx/ucx/tree/master/config/m4

**Comment by igor-ivanov** [comment]
Added other version with pkg.m4 as a part of sources

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/101/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/102/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/103/ for details (Mellanox internal link).

**Comment by Di0gen** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/3/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/120/ for details (Mellanox internal link).

**Comment by Di0gen** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/121/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/4/ for details (Mellanox internal link).

**Comment by Di0gen** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/5/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/122/ for details (Mellanox internal link).

**Comment by Di0gen** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/6/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/123/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/7/ for details (Mellanox internal link).

**Comment by miked-mellanox** [comment]
:+1: 

@igor-ivanov , @OphirMunk - what is a reason to keep auto-generated files config/* in the src tree?
IMO it should be removed.

https://github.com/Mellanox/libvma/tree/master/config

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/124/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
@miked-mellanox do you mean config.guess and config.sub files? There is related redmine issue issue :408075

**Comment by miked-mellanox** [comment]
i see. new autotools support arm. @rosenbaumalex - do we test arm? we will add arm soon to jenkins.

**Comment by Di0gen** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/9/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/126/ for details (Mellanox internal link).

**Pull request merged**